Slim/Control/Queries.pm

```diff
@@ -1043,10 +1043,18 @@ sub artistsQuery {
 			$roles = [ map { Slim::Schema::Contributor->typeToRole($_) } split(/,/, $roleID ) ];
 		}
 		elsif ($prefs->get('useUnifiedArtistsList')) {
-			$roles = Slim::Schema->artistOnlyRoles();
+			# include user-defined roles that user wants in artist list
+			my @udr;
+			foreach (keys %{$prefs->get('userDefinedRoles')}) {
+				push @udr, $_ if $prefs->get('userDefinedRoles')->{$_}->{'include'};
+			}
+			$roles = Slim::Schema->artistOnlyRoles(@udr);
```

Comment:
I think it's this construct which triggered the "move to schema" discussion initially, as it's being used +/- identically in search. Make it something like `getRolesToInclude()`?